sklearn/utils/validation.py

```diff
@@ -449,6 +449,16 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
     # DataFrame), and store them. If not, store None.
     dtypes_orig = None
     if hasattr(array, "dtypes") and hasattr(array.dtypes, '__array__'):
+        # throw warning if pandas dataframe is sparse
+        try:
+            array.sparse
+            warnings.warn(
+                "pandas Sparse Dataframe found."
+                "It will be inflated automatically."
+            )
+        except AttributeError:
```

Comment:
I wouldn't rely on DataFrame.sparse raising an attribute error. Users can have a column named `sparse`, and we currently let that override DataFrame accessors (probably a bug in pandas). More realistically, I could see DataFrame.sparse growing methods that work with non-sparse values (like a DataFrame.sparse.is_sprase)` that works with either a mix of dense / sparse, or all-sparse data.

Instead, I'd recommend applying `pd.api.types.is_sparse` to each column and checking if there are any / all sparse dtypes.

Reply:
Thank you for suggesting it!
I tried doing `df.dtypes.apply(pd.api.types.is_sparse).any()`. But it's not working properly for pandas versions "0.23.x". See, Linux py35_conda_openblas(pandas version 0.23.1) is failing. But Linux pylatest_pip_openblas_pandas(pandas version 0.25.3) passes.

Reply:
>I tried doing df.dtypes.apply(pd.api.types.is_sparse).any(). But it's not working properly for pandas versions "0.23.x". See, Linux py35_conda_openblas(pandas version 0.23.1) is failing. But Linux pylatest_pip_openblas_pandas(pandas version 0.25.3) passes.

Should I just put a condition for ignoring the test for pandas version < "0.24.0" or is there any other solution?